src/microbe_stage/Microbe.Contact.cs

```diff
@@ -383,16 +424,124 @@ public bool CanEngulf(Microbe target)
             return false;
         }
 
+        // Limit amount of things that can be engulfed at once
+        if (IngestedSizeCount > EngulfSize || IngestedSizeCount + target.EngulfSize > EngulfSize)
+            return false;
+
+        // Too many things attempted to be engulfed at once
+        if (attemptingToEngulf.Sum(e => e.EngulfSize) + target.EngulfSize > EngulfSize)
+            return false;
+
         // Disallow cannibalism
-        if (target.Species == Species)
+        if (isMicrobe && microbe!.Species == Species)
             return false;
 
         // Membranes with Cell Wall cannot engulf
         if (Membrane.Type.CellWall)
             return false;
 
         // Needs to be big enough to engulf
-        return EngulfSize >= target.EngulfSize * Constants.ENGULF_SIZE_RATIO_REQ;
+        return EngulfSize > target.EngulfSize * Constants.ENGULF_SIZE_RATIO_REQ;
+    }
+
+    public void OnAttemptedToBeEngulfed()
+    {
+        Membrane.WigglyNess = 0;
+
+        // Make the render priority of our organelles be on top of the highest possible render priority
+        // of the hostile engulfer's organelles
+        var hostile = HostileEngulfer.Value;
+        if (hostile != null)
+        {
+            foreach (var organelle in organelles!)
+            {
+                var newPriority = Mathf.Clamp(Hex.GetRenderPriority(organelle.Position) +
+                    hostile.organelles!.MaxRenderPriority, 0, Material.RenderPriorityMax);
```

Comment:
Found a possible cheaper way of doing this.

Reply:
I guess I need to wait for your next commit, but I'd like to clarify now that I meant creating a variable outside the loop starting on line 456, so basically just extract a local variable out of `hostile.organelles!.MaxRenderPriority`

Reply:
What I did was basically utilizing OnOrganelleAdded and OnOrganelleRemoved to calculate the max render priority with Mathf.Max.

Reply:
Okay, that makes sense as well as it is similar to like the toxin vacuole count variable. Though does it actually work for organelle remove in all cases? Because if the max value is removed, then you don't know it after that. I think perhaps a better solution would be to mark the value as dirty and call the expensive calculation method when the value is needed, that way removing organelles can still keep the value working, and not just always make it higher and higher.

Reply:
I think you're right.

Reply:
Should be good now.

Reply:
It follows `Microbe.membraneOrganellePositionsAreDirty` where `true` is the default value, but for updation I think `Microbe.HexCount` approach is a better example.

Reply:
Yeah, HexCount is the "proper" way I'd do it here. The reason why membrane organelle positions is like that is that it is just a flag for when the update logic needs to update the membrane, and not really the same thing logically as a variable that is cached when asked.